blackbird_python/blackbird/program.py

```diff
@@ -314,25 +313,46 @@ def serialize(self):
 
         script = ["name {}".format(self.name), "version {}".format(self.version)]
 
-        if self.target["name"] is not None:
-            array_insert += 1
-            options = ""
+        # add target and type to the script
+        for name, data in [("target", self.target), ("type", self.programtype)]:
```

Comment:
I found it difficult at first to parse this loop, may be worth checking how the code would look by adding target & type sequentially? Imho, a couple of lines duplicated lines could be fine

Reply:
Would it be nicer to instead add a separate function e.g. `self.add_metadata("target")` and `self.add_metadata("type")`?

Reply:
Looking at it again, I actually think I prefer this solution; without adding it sequentially (or creating separate functions).